modules/pam_mkhomedir/mkhomedir_helper.c

```diff
@@ -378,12 +375,16 @@ main(int argc, char *argv[])
 
    if (argc >= 3) {
 	char *eptr;
+	mode_t u_mask;
+
 	errno = 0;
 	u_mask = strtoul(argv[2], &eptr, 0);
 	if (errno != 0 || *eptr != '\0') {
 		pam_syslog(NULL, LOG_ERR, "Bogus umask value %s", argv[2]);
 		return PAM_SESSION_ERR;
 	}
+
+	umask(u_mask);
```

Comment:
Please call umask with the default 0022 value when the umask argument is not specified.